lib/nuget-parser/csproj-parser.ts

```diff
@@ -5,16 +5,12 @@ import * as path from 'path';
 import * as parseXML from 'xml2js';
 import * as _ from 'lodash';
 import * as debugModule from 'debug';
+import { TargetFramework } from './types';
+import { toReadableFramework } from './framework';
 const debug = debugModule('snyk');
 
-interface TargetFramework {
-  framework: string;
-  original: string;
-  version: string;
-}
-
-export async function getTargetFrameworksFromProjFile(rootDir: string): Promise<any> {
-  return new Promise((resolve, reject) => {
+export async function getTargetFrameworksFromProjFile(rootDir: string): Promise<TargetFramework | undefined> {
+  return new Promise<TargetFramework | undefined>((resolve, reject) => {
```

Comment:
probably no longer need this whole `new Promise` just use async / await directly and throw?

Reply:
Was hoping for the same, but the `parseXml` function actually expects a callback function, hence why it was wrapped in a Promise to begin with. The callback function gets called whenever `parseXml` is done, so we can't await on it (there is no interface with a Promise :/)

Reply:
wanted to suggest using native `util` module's `promisify` method available from Node 8: 
`const parseXML = util.promisify(parseXML.parseString);`
then you could use `async / await`